File: modules/shared_services/shared-services/src/DataBuilder/BrokerDataBuilder.php
<?php

/**
 * This file is part of the Spryker Suite.
 * For full license information, please view the LICENSE file that was distributed with this source code.
 */

namespace SharedServices\DataBuilder;

class BrokerDataBuilder implements SharedServiceDataBuilderInterface
{
    public function build(array $data): array
    {

        return $data;
//        todo: should be fixed
        $api = $data['api'] ?? [];

        if (empty($api)) {
            return $api;
        }

        $username = $api['username'] ?? [];
        $password = $api['password'] ?? [];

        return  [
            'api' => [
                'username' => array_unique($username),
                'password' => array_unique($password),
            ],
        ];
    }

    public function getSharedServiceName(): string
    {
        return 'broker';
    }
}

File: modules/project_manager/project_manager/project/data_builder/broker_data_builder.py
from project_manager.project.data_builder.data_builder_interface import DataBuilderInterface


class BrokerBuilder(DataBuilderInterface):
    def get_service_name(self) -> str:
        return 'broker'

    def build(self, project_data: dict, shared_service_data: dict) -> dict:
        if 'api' not in shared_service_data:
            return {}

        shared_service_data['data'] = {
            'api': shared_service_data['api'],
        }

        return shared_service_data
